### ISSUE TITLE
!saved_model_cli run command does not allow placeholder for --dir

### ISSUE THREAD

**Issue by utibeng** [tensorflow/tensorflow#55400]
On Google Colab, I can run **!saved_model_cli show --dir {serving_location}**  having specified serving_location as a variable and the command yields the expected result.

However, when I try running **!saved_model_cli run --dir {serving_location}**, I receive the error 
OSError: SavedModel file does not exist at: {serving_location}.

I can successfully run the command by replacing the {serving_location} placeholder with the actual path string e.g. /content/drive.

tf version is 2.8.0

**Comment by tilakrayal** [comment]
@utibeng ,
Can you please take a look at this link [1](https://github.com/tensorflow/tensorflow/issues/22480), [2](https://github.com/tensorflow/serving/issues/491) and [3](https://www.tensorflow.org/guide/saved_model) which delivers the information.It helps.Thanks!

**Comment by utibeng** [comment]
Yes I have looked at them, but the issues are kind of different. In my case, I can replace the placeholder with the actual string and it works. So I am pointing at the right folder

**Comment by tilakrayal** [comment]
@utibeng ,
 In order to expedite the trouble-shooting process, could you please provide a complete code you are using.

**Comment by utibeng** [comment]
I have a folder containing a trained keras model and the assets. Pretty much identical to the standard keras saved model. The folder's location is /home/aa/bb/. The folder contains;
 - folder assets
 - folder variables
 - saved_model.pb
 - keras_metadata.pb

then i run the following commands and get the outputs

**serving_location = ' /home/aa/bb/'
!saved_model_cli show --dir {serving_location} --tag_set serve --signature_def transform_feature**

Output is as expected:
The given SavedModel SignatureDef contains the following input(s):
  inputs['examples'] tensor_info:
      dtype: DT_STRING
      shape: (-1)
      name: transform_features_examples:0
The given SavedModel SignatureDef contains the following output(s):
  outputs['Age'] tensor_info:
      dtype: DT_FLOAT
      shape: (-1)
      name: StatefulPartitionedCall_26:0
  outputs['Embarked'] tensor_info:
      dtype: DT_INT64
      shape: (-1)
      name: StatefulPartitionedCall_26:1
  outputs['Fare'] tensor_info:
      dtype: DT_FLOAT
      shape: (-1)
      name: StatefulPartitionedCall_26:2
  outputs['Parch'] tensor_info:
      dtype: DT_INT64
      shape: (-1)
      name: StatefulPartitionedCall_26:3
  outputs['Pclass'] tensor_info:
      dtype: DT_INT64
      shape: (-1)
      name: StatefulPartitionedCall_26:4
  outputs['Sex'] tensor_info:
      dtype: DT_INT64
      shape: (-1)
      name: StatefulPartitionedCall_26:5
  outputs['SibSp'] tensor_info:
      dtype: DT_INT64
      shape: (-1)
      name: StatefulPartitionedCall_26:6
  outputs['Survived'] tensor_info:
      dtype: DT_INT64
      shape: (-1)
      name: StatefulPartitionedCall_26:7
Method name is: tensorflow/serving/predict

Next I run the command
**!saved_model_cli run --dir {serving_location} \
--tag_set serve --signature_def transform_features --input_examples \
'examples=[{"Sex":["male"],"Age":[17.0], "Fare":[8.05],"Name":["emilitus"], "Parch":[0] \
, "PassengerId":[88888], "SibSp":[1], "Survived":[999999],"Ticket":["373450"] \
, "Cabin":[" "]'}]** 
the above fails with the message below
Traceback (most recent call last):
  File "/usr/local/bin/saved_model_cli", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/tensorflow/python/tools/saved_model_cli.py", line 1260, in main
    args.func(args)
  File "/usr/local/lib/python3.7/dist-packages/tensorflow/python/tools/saved_model_cli.py", line 767, in run
    init_tpu=args.init_tpu, tf_debug=args.tf_debug)
  File "/usr/local/lib/python3.7/dist-packages/tensorflow/python/tools/saved_model_cli.py", line 415, in run_saved_model_with_feed_dict
    tag_set)
  File "/usr/local/lib/python3.7/dist-packages/tensorflow/python/tools/saved_model_utils.py", line 113, in get_meta_graph_def
    saved_model = read_saved_model(saved_model_dir)
  File "/usr/local/lib/python3.7/dist-packages/tensorflow/python/tools/saved_model_utils.py", line 51, in read_saved_model
    raise IOError("SavedModel file does not exist at: %s" % saved_model_dir)
OSError: SavedModel file does not exist at: {serving_location}

to demonstrate that is failing because of the location placeholder, I run the same command but specify the actual location

**!saved_model_cli run --dir /home/aa/bb \
--tag_set serve --signature_def transform_features --input_examples \
'examples=[{"Sex":["male"],"Age":[17.0], "Fare":[8.05],"Name":["emilitus"], "Parch":[0] \
, "PassengerId":[88888], "SibSp":[1], "Survived":[999999],"Ticket":["373450"] \
, "Cabin":[" "]'}]** 

2022-04-02 18:52:52.501280: E tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/tensorflow/python/tools/saved_model_cli.py:451: load (from tensorflow.python.saved_model.loader_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.loader.load or tf.compat.v1.saved_model.load. There will be a new function for importing SavedModels in Tensorflow 2.0.
INFO:tensorflow:Restoring parameters from /content/drive/MyDrive/AI_2020/kaggle/titanic/serving_model/titanic-pipeline/1648397864/variables/variables
Result for output key Age:
[35.]
Result for output key Embarked:
[0]
Result for output key Fare:
[8.05]
Result for output key Parch:
[0]
Result for output key Pclass:
[0]
Result for output key Sex:
[0]
Result for output key SibSp:
[0]
Result for output key Survived:
[999999]


So basically the !saved_model_cli run --dir command does not work when we use a placeholder to specify the directory, but the !saved_model_cli show --dir works.

**Comment by tilakrayal** [comment]
@utibeng ,
This issue is more related to Tensorflow/Serving repo.I request you, please post in Serving repo from [here](https://github.com/tensorflow/serving/issues).
